pilot/pkg/model/sidecar.go

```diff
@@ -477,15 +477,25 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 			for _, importedHost := range importedHosts {
 				// Check if the hostnames match per usual hostname matching rules
 				if importedHost.Matches(s.Hostname) {
-					// TODO: See if the service's ports match.
-					//   If there is a listener port for this Listener, then
-					//   check if the service has a port of same value.
```

Comment:
Couple of points here on the TODO
-  If not, check if the service has a single port - and choose that port - Why should a service be selected if it has single non matching port?
- I have left a TODO to check if there is VS with the matching port , for now if service does not have matching ports, it just drops the service. I plan to implement this in a follow-up.

Reply:
because the egress is effectively a http proxy of sorts. eg I could egress out of 9080 but go to a destination service on 7080.

Reply:
Thank you. That makes sense - But why are we applying that logic if the service has only single port? Why not if it has multiple non matching ports with out any virtual services defined - Should not we select that service based on the same logic? And if we do - does n't it negate port filtering completely?

Reply:
The port is not a matching rule. After this, will it influence the http proxy as described above?

Reply:
If port  is not a matching rule, how do we say to Pilot to not create these upstream clusters if we do not want http proxy behaviour? i.e. only create clusters and routes with matching port given in egress listener?

Reply:
Also looking at the port documentation "The port if specified, will be used as the default destination port associated with the imported hosts." - Does n't it mean if an egress listener is specified with a port - it should use the same port on the service? Or Am I understanding this incorrectly?

Reply:
> Thank you. That makes sense - But why are we applying that logic if the service has only single port? Why not if it has multiple non matching

So if a service has only one port 7080, then when doing an egress from 9080, we can confidently route to the service on 7080 even without virtual service.  But if a service has multiple ports, then we have no way of knowing which port to send to, without a virtual service. If there is a virtual service then, we expect the destination in the virtual service to tell us what port should be used. 

If you feel that this is too complicated, we can use the logic that you proposed (egress on 9080 will only filter services with 9080 port). I don’t think very many people are using the egress with ports. But if we are doing this, we should guard this with a flag for this release, change api docs, etc. and then remove the current behavior in next release permanently. 

@frankbu / @louiscryan wdyt about ^^^ semantics?

Reply:
> Also looking at the port documentation "The port if specified, will be used as the default destination port associated with the imported hosts." - Does n't it mean if an egress listener is specified with a port - it should use the same port on the service? Or Am I understanding this incorrectly?

Hmm. I guess if we have already declared it this way, then we could change the implementation

Reply:
Yeah. Since we already have documented it like that - I think it is OK to implement it like that. This PR already does that. Only VS matching is pending - I can implement as part of this or as follow-up based on what people think.

Reply:
@rshriram Implemented Virtual service matching as well. Now it matches with what docs says. PTAL if this makes sense.